### PR TITLE
93/Implemented reusable table component for angular application.

### PR DESCRIPTION
**Pull request by rajat-vrma**
Issue: [93](https://github.com/Innostax/tavajs/issues/93)

### CODE REVIEW

**Review by manjeetbeniwalinnostax — APPROVED** [review]

**Comment by manjeetbeniwalinnostax** [comment]
will resolve comments in the follow up PR

File: src/templates/angular/src/app/pages/users/users.component.ts [hunk 1]
```diff
@@ -31,19 +37,34 @@ export class UsersComponent implements OnInit {
...
-      this.users = res
+      this.users = res.map((each: any)=> {
+        const refactoredUserData: object = {
+          id: each.id,
+          name: each.name,
+          username: each.username,
+          email: each.email,
+        }
+        return refactoredUserData
+        }
```
Comment (Chaman-Raghav) [inline src/templates/angular/src/app/pages/users/users.component.ts:48]: this.users = res.map((each: any)=> ({
          id: each.id,
          name: each.name,
          username: each.username,
          email: each.email,
        })

File: src/templates/angular/src/app/pages/users/users.component.ts [hunk 2]
```diff
@@ -31,19 +37,34 @@ export class UsersComponent implements OnInit {
...
   
+  <% if(isCrud || isCrudWithNode){%>
   onClickAddUser() {
     $('#addUser_modal').modal('show');
     this.data = {};
   }
 
+  public handleUserActions = (name: any, user: any) => {
+    if(name == EDIT) this.editUser(user)
+    else this.deleteUser(user.id)
```
Comment (Chaman-Raghav) [inline src/templates/angular/src/app/pages/users/users.component.ts:62]: on else delete?
is there only two possible operation, lets specify name as well for delete, we shouln't delete in some else block, without specify operation action

File: src/templates/angular/src/app/shared/components/molecules/table/table.component.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  styleUrls: ['./table.component.css']
+})
+export class TableComponent {
+  @Input() usersData: any;
+  @Input() public handleEditDeleteAction!: (name:any, user:any) => void;
+  @Input() shouldShowActions: boolean = false;
+  @Input() headers: any;
+
+  data: any;
+  public keepOriginalOrder = (a:any, b:any) => a.key;
```
Comment (Chaman-Raghav) [inline src/templates/angular/src/app/shared/components/molecules/table/table.component.ts:17]: @manjeetbeniwal 
Let's avoid any type going forward.

File: src/templates/angular/src/app/shared/components/molecules/table/table.component.ts [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes['usersData']?.currentValue) {
+      this.data = this.usersData;
+    }
+  }
+
+  handleUser(name:any, user:any) {
+    this.handleEditDeleteAction(name,user);
```
Comment (Chaman-Raghav) [inline src/templates/angular/src/app/shared/components/molecules/table/table.component.ts:26]: formatting is missing, please integrate eslinting etc

**Review by Chaman-Raghav — COMMENTED** [review]
Approved with suggestions

**Pull request merged**